api/api/version_endpoints_api.go

```diff
@@ -251,6 +256,11 @@ func (c *EndpointsController) UpdateEndpoint(r *http.Request, vars map[string]st
 			}
 		}
 
+		// Should not allow redeploying a "serving" model and changing its deployment type since we can't guaratee graceful deployment
+		if endpoint.IsServing() && newEndpoint.DeploymentType != endpoint.DeploymentType {
```

Comment:
So we allow the redeployment if:
1. Endpoint status is not serving so we can change the deployment type
2. Endpoint status is serving and we do not change the deployment type

Reply:
Yes